org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -630,4 +640,100 @@ public void reportProjectsStatus() {
 		}
 	}
 
+	class UpdateProjectsWorkspaceJob extends WorkspaceJob {
+
+		private final Collection<IProject> projects;
+		private final boolean force;
+
+		public UpdateProjectsWorkspaceJob(Collection<IProject> projects, boolean force) {
+			super("Updating project configurations");
+			this.projects = projects;
+			this.force = force;
+		}
+
+		@Override
+		public boolean belongsTo(Object family) {
+			return IConstants.UPDATE_PROJECT_FAMILY.equals(family) || IConstants.JOBS_FAMILY.equals(family);
+		}
+
+		@Override
+		public IStatus runInWorkspace(IProgressMonitor monitor) {
+			long start = System.currentTimeMillis();
+			MultiStatus status = new MultiStatus(IConstants.PLUGIN_ID, 0, "Update project configurations");
+			for (Entry<String, Map<IBuildSupport, IProject>> entry : groupByBuildToolName(projects).entrySet()) {
+				Map<IBuildSupport, IProject> projectsWithBuildSupport = entry.getValue();
+				String buildToolName = entry.getKey();
+				if (!Objects.equals(buildToolName, "UnknownBuildTool")) {
```

Comment:
I suggest removing this check since invoking onWillUpdate on ‘UnknownBuildTool’ has no effect and can simplify the codebase.

Reply:
Ok

Reply:
Wait, I remembered why I added this check.

If two different build support implementations do not override `buildToolName()` but have their own implementation of `onWillUpdate()`, then they will be groupped into one entry and one build support will miss the invocation of `onWillUpdate()`

Reply:
I disagree. If that's said, onWillUpdate is never invoked for them.

I think you should not group "UnknownBuildTool" as one group, but keep them as individual.

Reply:
And you can group them by the build support class instance instead of buildToolName.